src/zcash/NoteEncryption.cpp

```diff
@@ -187,6 +187,43 @@ boost::optional<SaplingEncPlaintext> AttemptSaplingEncDecryption(
     return plaintext;
 }
 
+boost::optional<SaplingEncPlaintext> AttemptSaplingEncDecryption (
+    const SaplingEncCiphertext &ciphertext,
+    const uint256 &epk,
+    const uint256 &esk,
+    const uint256 &pk_d
```

Comment:
This is confusing; is it the version of note ciphertext decryption that is only used when decrypting using a full viewing key? Why isn't it named that way?

In general I'm having difficulty following the code in this PR because it is structured very differently to the spec. (That's probably why I didn't finish reviewing it before it was merged.)

Reply:
@daira Yes, it was named with full viewing key, but then changed.  Reason here: https://github.com/zcash/zcash/pull/3391#discussion_r204415496

Reply:
I think this was an overuse of overloading, but yes we can refactor later.

Reply:
I still don't like this name, but yes we can refactor later.